ckan/lib/search/index.py

```diff
@@ -245,6 +245,8 @@ def index_package(self, pkg_dict, defer_commit=False):
                         value = None
                 except ValueError:
                     continue
+                except TypeError:
```

Comment:
Again I think this may be masking a problem elsewhere.  I'm not suggesting that we shouldn't handle all cases - but given the previous comment - perhaps it would be more straight-forward to find the timezone on the date being passed in oaipmh.harvester?